### PR TITLE
Fix a bug in gx integration

### PR DESCRIPTION
**Pull request by XinEDprob**
# TL;DR
Fixed a bug in the integration of Flyte and Great Expectations. Without this fix, pyspark dataframe will always be converted to pandas dataframe in Great Expectations. As a result, there will have an error when people want to use `SparkDFExecutionEngine`.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [] Any pending items have an associated Issue

## Complete description
I use `selected_datasource[0]["execution_engine"]["class_name"]` to recognize the `execution_engine` declared by the user for Great Expectations. If it is a `SparkDFExecutionEngine`, the data which is in the form FlyteSchema is converted to `pyspark.sql.dataframe.DataFrame`, else transform the data to the default `pandas.dataframe`.

## Tracking Issue
[https://github.com/flyteorg/flyte/issues/<number>](https://github.com/flyteorg/flyte/issues/3739)

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by XinEDprob** [comment]
@pingsutw thanks for letting me know. Will take a look at the test and finish the corresponding change in this week.

**Review by pingsutw — APPROVED** [review]

File: plugins/flytekit-greatexpectations/flytekitplugins/great_expectations/task.py [hunk 1]
```diff
@@ -183,10 +184,16 @@ def execute(self, **kwargs) -> Any:
             if is_runtime and issubclass(datatype, str):
                 final_batch_request["runtime_parameters"]["query"] = dataset
             elif is_runtime and issubclass(datatype, FlyteSchema):
-                final_batch_request["runtime_parameters"]["batch_data"] = dataset.open().all()
+                # if execution engine is SparkDF, transform the data to pyspark.sql.dataframe.DataFrame, else transform the data
+                # to the default pandas.dataframe
+                if selected_datasource[0]["execution_engine"]["class_name"] == "SparkDFExecutionEngine":
+                    final_batch_request["runtime_parameters"]["batch_data"] = dataset.open(
```
Comment (pingsutw) [inline plugins/flytekit-greatexpectations/flytekitplugins/great_expectations/task.py:190]: nit: Could we move import to here? So current users don't necessarily have to install pyspark.
```suggestion
                if selected_datasource[0]["execution_engine"]["class_name"] == "SparkDFExecutionEngine":
                    import pyspark
                    final_batch_request["runtime_parameters"]["batch_data"] = dataset.open(
```

Comment (XinEDprob) [reply plugins/flytekit-greatexpectations/flytekitplugins/great_expectations/task.py]: good point, just moved pyspark.

**Review by pingsutw — DISMISSED** [review]

**Review by pingsutw — COMMENTED** [review]
There is a lint error, could you run `make lint` and push it again?

image

**Review by pingsutw — APPROVED** [review]

**Comment by pingsutw** [comment]
@XinEDprob nvm, that's a flaky test. merged it
